src/lib_json/json_reader.cpp

```diff
@@ -1873,20 +1868,36 @@ std::vector<OurReader::StructuredError> OurReader::getStructuredErrors() const {
 }
 
 class OurCharReader : public CharReader {
+public:
   bool const collectComments_;
   OurReader reader_;
 
-public:
   OurCharReader(bool collectComments, OurFeatures const& features)
-      : collectComments_(collectComments), reader_(features) {}
-  bool parse(char const* beginDoc, char const* endDoc, Value* root,
-             String* errs) override {
-    bool ok = reader_.parse(beginDoc, endDoc, *root, collectComments_);
-    if (errs) {
-      *errs = reader_.getFormattedErrorMessages();
+      : collectComments_(collectComments), reader_(features) {
+        this->_impl = std::unique_ptr<OurCharReader::Impl>(new OurCharReader::Impl(collectComments, features));
+      }
+
+  class Impl : public CharReader::Impl {
+  public:
+    bool const collectComments_;
```

Comment:
I think the convention here is for data members to be placed at the bottom of the class def

Reply:
Ok, this one wasn't done. Can you please move these data members to the bottom of the class, after the member functions?